VEDA_OS01/management/commands/create_oauth_client.py

```diff
@@ -0,0 +1,143 @@
+        )
+        parser.add_argument(
+            '-r',
+            '--redirect_uris',
+            help="Comma separated redirect URIs."
+        )
+        parser.add_argument(
+            '-n',
+            '--client_name',
+            help="String to assign as the Client name."
+        )
+        parser.add_argument(
+            '-s',
+            '--client_secret',
+            help="String to assign as the Client secret. Should not be shared."
+        )
+        parser.add_argument(
+            '-a',
+            '--skip_authorization',
+            action='store_true',
+            default=False,
+            help="Skip authorization for trusted applications."
+        )
+
+    def handle(self, *args, **options):
+        self._clean_required_args(options['client_id'], options['client_type'], options['authorization_grant_type'])
+        self._parse_options(options)
+
+        client_id = self.fields.get('client_id')
+
+        # Check if client ID is already in use. If so, fetch existing Client and update fields.
+        Application.objects.update_or_create(
+            client_id=self.fields.pop('client_id'), defaults=self.fields
+        )
+
+    def _clean_required_args(self, client_id, client_type, grant_type):
+        """
+        Validate and clean the command's arguments.
+
+        Arguments:
+            client_id (str): Client Id.
+            client_type (str): Client Type.
+            grant_type (str): Grant Type
+
+        Raises:
+            CommandError, if the arguments have invalid values.
+        """
+        client_id = client_id and client_id.strip()
+        if not client_id:
+            raise CommandError('Client id provided is invalid.')
+
+        client_type = client_type.lower()
+        if client_type not in CLIENT_TYPES:
+            raise CommandError('Client type provided is invalid. Please use one of {}.'.format(CLIENT_TYPES))
+
+        grant_type = grant_type.lower()
+        if grant_type not in GRANT_TYPES:
+            raise CommandError('Grant type provided is invalid. Please use one of {}.'.format(GRANT_TYPES))
+
+        self.fields = {  # pylint: disable=attribute-defined-outside-init
+            'client_id': client_id,
+            'client_type': client_type,
+            'authorization_grant_type': grant_type,
+        }
+
+    def _parse_options(self, options):
+        """
+        Parse the command's options.
+
+        Arguments:
+            options (dict): Options with which the command was called.
+
+        Raises:
+            CommandError, if user does not exist or redirect_uris is invalid
+        """
+        for key in ('username', 'client_name', 'client_secret', 'redirect_uris', 'skip_authorization'):
+            value = options.get(key)
+            if value is not None:
+                self.fields[key] = value
+
+        username = self.fields.pop('username', None)
+        if username is not None:
```

Comment:
We are pushing `user_name` then we're popping and doing stuff and again mapping the `user_id` in `fields`. Same goes for `client_name`, we should do all these in the above main loop on L#118?